frontend/src/components/AppSelect.vue

```diff
@@ -80,81 +85,151 @@ import {
+        /** limit popover height to available height */
         maxHeight: `${availableHeight - 20}px`,
       });
     },
   }),
 ];
 
-// model value to pass from parent to headlessui
-const value = computed(() =>
-  props.multi
-    ? props.options.filter((option) => props.modelValue.includes(option.id))
-    : props.options.find((option) => option.id === props.modelValue) || "",
-);
+/** normalize single/multi to array */
+function normalize<T>(value: T | T[]): T[] {
+  return Array.isArray(value) ? value : [value];
+}
+
+/** model value to pass from parent to headlessui */
+const value = computed(() => {
+  let list = normalize(props.modelValue);
+  return props.multi
+    ? props.options.filter((option) => list.includes(option.id))
+    : props.options.find((option) => list.includes(option.id)) || "";
+});
 
-// model value to emit from headlessui to parent
-async function onChange(value: Option | Option[]) {
-  const id = Array.isArray(value) ? value.map((v) => v.id) : value.id;
-  // https://github.com/ycs77/headlessui-float/issues/80
-  await sleep(10);
+/** model value to emit from headlessui to parent */
+async function onChange(value: O | O[]) {
+  let list = normalize(value);
+  const id = props.multi ? list.map((v) => v.id) : list[0]?.id || "";
   emit("update:modelValue", id);
 }
 
-// label to show as selected value in box
+/** full selected option */
+const selectedOption = computed(() => {
+  /** normalize to array */
+  let list = normalize(props.modelValue);
+  if (!props.multi)
+    return props.options.find((option) => option.id === list[0]);
+  else return undefined;
+});
+
+/** label to show as selected value in box */
 const selectedLabel = computed<string>(() => {
-  const value = props.multi
-    ? props.options
-        .filter((option) => props.modelValue.includes(option.id))
-        .map((option) => option.label)
-    : props.options.find((option) => option.id === props.modelValue)?.label ||
-      "";
-  if (!Array.isArray(value)) return value;
+  /** normalize to array */
+  let list = normalize(props.modelValue);
+  if (!props.multi)
+    return (
+      props.options.find((option) => option.id === list[0])?.label ||
+      "None selected"
+    );
+  const value = props.options.filter((option) => list.includes(option.id));
   if (value.length === 0) return "None selected";
-  if (value.length === 1) return value[0] || "";
+  if (value.length === 1) return value[0]?.label || "1 Selected";
   if (value.length === props.options.length) return "All selected";
   return value.length + " selected";
 });
+
+/** whether all options are selected */
+const allSelected = computed(
+  () => props.multi && props.options.length === props.modelValue.length,
+);
+
+/** select/deselect all */
+function toggleAll() {
+  emit(
+    "update:modelValue",
+    allSelected.value ? [] : props.options.map((option) => option.id),
+  );
+}
+
+/** add "quick" arrow key select */
+function onKeypress({ key }: KeyboardEvent) {
```

Comment:
When button is focused, left/right arrow keys will quickly scroll through the options without opening the dropdown. Up/down opens the dropdown (via headless ui), per expected accessible select behavior.